### PR TITLE
[WIP] Add option to load emotion without ajax

### LINKED ISSUE #11: Load emotions without ajax
https://github.com/shopware/shopware/commit/2720fa13520b6b49f56fb94696ca2937e845749d

Maybe interesting? :-)

### PR DESCRIPTION
**Pull request by tinect**
Took new Feature from 5.6.
fixes #11 

Please review :-)

### CODE REVIEW

File: Resources/views/frontend/_public/js/jquery.emotion.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+$.overridePlugin('swEmotionLoader', {
```
Comment (PaddyS) [inline Resources/views/frontend/_public/js/jquery.emotion.js]: Dangerous.
You would have to keep track of every change from shopware itself in order to not re-include bugs.

Just as an example, [this](https://github.com/shopware/shopware/pull/1605/files) was **almost** merged into shopware and would have been reopened with your changes.

**Comment by PaddyS** [comment]
I like that idea in general, but:
I'd add a condition for all of those changes to only take place with Shopware version < 5.6.

Just to be sure you're not reverting bugfixes that have been made in Shopware 5.6.x itself. :-)

**Comment by tinect** [comment]
> I like that idea in general, but:
> I'd add a condition for all of those changes to only take place with Shopware version < 5.6.
> 
> Just to be sure you're not reverting bugfixes that have been made in Shopware 5.6.x itself. :-)

Yes, there are many features in this plugin, implemented in SW5.6.

**Comment by shyim** [comment]
Maybe we add a second views folder, which will be loaded for that

File: Resources/views/frontend/_public/js/jquery.emotion.js (hunk 1 above)
Comment (tinect) [reply Resources/views/frontend/_public/js/jquery.emotion.js]: Damn. So true.

**Comment by tinect** [comment]
Just noticed, that ALL images of every emotion will be loaded - no matter if emotion has `display:none` 💭

File: Subscriber/TemplateSubscriber.php [hunk 2]
```diff
@@ -14,14 +16,46 @@ class TemplateSubscriber implements SubscriberInterface
...
+
     /**
      * TemplateSubscriber constructor.
      *
      * @param $viewDir
+     * @param $emotionNoAjaxViewDir
+     * @param CachedConfigReader $configReader
      */
-    public function __construct($viewDir)
+    public function __construct($viewDir, $emotionNoAjaxViewDir, $version, CachedConfigReader $configReader)
```
Comment (shyim) [inline Subscriber/TemplateSubscriber.php]: INTERFACE

Comment (tinect) [reply Subscriber/TemplateSubscriber.php]: 😚

File: Subscriber/TemplateSubscriber.php [hunk 3]
```diff
@@ -14,14 +16,46 @@ class TemplateSubscriber implements SubscriberInterface
...
+    /**
+     * @var boolean
+     */
+    private $emotionPreLoading;
+
     /**
      * TemplateSubscriber constructor.
      *
      * @param $viewDir
+     * @param $emotionNoAjaxViewDir
```
Comment (PaddyS) [inline Subscriber/TemplateSubscriber.php]: TypeHints in Docs :-)

File: Subscriber/TemplateSubscriber.php [hunk 4]
```diff
@@ -14,14 +16,46 @@ class TemplateSubscriber implements SubscriberInterface
...
      */
-    public function __construct($viewDir)
+    public function __construct($viewDir, $emotionNoAjaxViewDir, $version, ConfigReader $configReader)
     {
         $this->viewDir = $viewDir;
+        $this->emotionNoAjaxViewDir = $emotionNoAjaxViewDir;
+        $this->version = $version;
+        $this->configReader = $configReader;
+
+        if(version_compare($this->version, '5.6.0', '>=')) {
```
Comment (PaddyS) [inline Subscriber/TemplateSubscriber.php]: Whitespace

**Pull request closed without merging**